src/dune_rules/ordered_set_lang.ml

```diff
@@ -314,6 +314,17 @@ module Unexpanded = struct
     in
     loop t.ast Pos init
 
+  let exists t ~f =
+    let rec loop (t : ast) =
+      let open Ast in
+      match t with
+      | Standard | Include _ -> false
+      | Element x -> f x
+      | Union l -> List.exists l ~f:loop
+      | Diff (l, f) -> loop l <> loop f
```

Comment:
The `Diff` case doesn't seem right.

Looking at this again, it's ok to use `fold_strings` and just raise a `Found` exception if you find the flag in Positive position. Reimplementing this logic isn't trivial.

Reply:
OK, I've just decided to check the expanded version instead as that is much simpler and more correct. We want to also check the --explain flag appearing in the env stanza. Therefore i have added an extra test case demonstrating --explain given from an env stanza.